wasm/wat/types.go

```diff
@@ -0,0 +1,83 @@
+package wat
+
+import "fmt"
+
+// module corresponds to the text format of a WebAssembly module, and is an intermediate representation prior to
+// wasm.Module. This is primarily needed to resolve symbolic indexes like "$main" to raw numeric ones.
+//
+// Note: nothing is required per specification. Ex `(module)` is valid!
+//
+// See https://www.w3.org/TR/wasm-core-1/#functions%E2%91%A7
+type module struct {
+	// name is optional and starts with '$'. For example, "$test".
+	// See https://www.w3.org/TR/wasm-core-1/#modules%E2%91%A0%E2%91%A2
+	//
+	// Note: The name may also be stored in the wasm.Module CustomSection under the key "name" subsection 0.
+	// See https://www.w3.org/TR/wasm-core-1/#binary-namesec
+	name string
+
+	// importFuncs are imports describing functions "(import... (func...))" added in insertion order.
+	importFuncs []*importFunc
+
+	// startFunction is the function to call during wasm.Store Instantiate. The value is a importFunc.name, such as
+	// "$main", or its equivalent raw numeric index, such as "2".
```

Comment:
wanna confirm that this is the reason why we have string here right?

Reply:
at the moment, yes. A future refactoring maybe we can eagerly map to an integer and drop the name, but we should do that after we start encoding the name custom section.